scripts/run_formatters

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+
+# Runs clang-format and rustfmt on the entire codebase.
+# Use -a option to apply the formattings, otherwise the script just runs a check.
+# Exit codes if option -a is not present:
+#  - 1 there are files to be formatted.
+#  - 0 everything looks fine.
+# If -a is set, the return code is always 0.
+
+# Parse optionns (-h and -a).
+APPLY_FORMATTING=false
+while getopts "ha" opt; do
+  case "$opt" in
+  h|\?)
+    echo "Use -a option to apply the formatting."
+    exit 0
+    ;;
+  a)
+    APPLY_FORMATTING=true
+    ;;
+  esac
+done
+
+# Get CPU count.
+OS=$(uname)
```

Comment:
Does the parallel processing actually save any nontrivial amount of time? I would be inclined to leave this complexity out unless there is a definite advantage.

Reply:
It would only ever help if the code grew to 100s of files probably. I removed it.